src/components/inputs/LeaseSelectInput.js

```diff
@@ -22,27 +31,86 @@ const LeaseSelectInput = ({
   onBlur,
   onChange,
   placeholder,
-  relatedLeases,
+  leaseHistoryItems,
   value,
 }: Props) => {
-  const getLeaseOptions = (leases: Array<Object>): Array<Object> =>
-    leases
-      .filter((lease) => relatedLeases.find((relatedLease) => lease.id === relatedLease.lease.id) ? false : true)
+  const leaseAttributes = getLeaseAttributes(store.getState());
+  const getHistoryItemOptions = (leases: Array<Object>, plotSearches: Array<Object>,plotApplications: Array<Object>, areaSearches: Array<Object>): Array<Object> => {
+    const filterOutExisting = (item) => leaseHistoryItems.find((leaseHistoryItem) => item.id === leaseHistoryItem.lease.id) ? false : true
+        
+    leases = leases
+      .filter(filterOutExisting)
       .map((lease) => {
+        const stateOptions = getFieldOptions(leaseAttributes, LeaseFieldPaths.STATE)
         return {
           value: lease.id,
-          label: getContentLeaseIdentifier(lease),
+          label: `${getContentLeaseIdentifier(lease) || ''}, ${getLabelOfOption(stateOptions, lease.state) || ''}`,
+          type: 'lease'
         };
       });
 
-  const getLeases = debounce(async(inputValue: string, callback: Function) => {
+    plotSearches = plotSearches
+      .filter(filterOutExisting)
+      .map((plotSearch) => {
+        const maxLength = 24
+        const displayName = plotSearch.name.length > maxLength ? plotSearch.name.substr(0,maxLength) + "..." : plotSearch.name
+        return {
+          value: plotSearch.id,
+          label: `${displayName}, Haku`,
```

Comment:
Get "Haku" from translations

Reply:
Something I noticed from the emails was this point which is moot, as the preparer UI is only in Finnish and does not have translations for its strings. (It is also hard coded to add an Accept-Language header to all API requests to retrieve data in translatable Django models in Finnish whenever possible.)